src/playsim/p_map.cpp

```diff
@@ -2065,15 +2065,20 @@ void P_FakeZMovement(AActor *mo)
 	{ // float down towards target if too close
 		if (!(mo->flags & MF_SKULLFLY) && !(mo->flags & MF_INFLOAT))
 		{
-			double dist = mo->Distance2D(mo->target);
+			double dist_squared = mo->Distance2DSquared(mo->target);
+			//still need delta intact, because it can be negative
+			//and square of a real number always positive
 			double delta = mo->target->Center() - mo->Z();
-			if (delta < 0 && dist < -(delta * 3))
+			//why it multiplies by 3 in original function?
+			double delta_squared = delta * abs(delta) * 3 * 3;
+			if (delta_squared < 0 && dist_squared < -(delta_squared) )
 				mo->AddZ(-mo->FloatSpeed);
-			else if (delta > 0 && dist < (delta * 3))
+
```

Comment:
please drop this empty line

Reply:
I wanted to group "different" operations in one "block", but Okay(3)